subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ScriptExecuter.groovy

```diff
@@ -34,6 +34,7 @@ class ScriptExecuter {
         } else {
             executable = "${workingDir}/${executable}"
         }
+        builder.environment("JAVA_HOME", System.getProperty("java.home"))
```

Comment:
This is actually always running tests with `java` in `PATH`, not the expected java version.